slither/slither_vyper.py

```diff
@@ -0,0 +1,151 @@
+        self._triage_mode = False
+        super(Slither, self).__init__(filename)
+
+        self._init_from_vyper(filename)
+        self._analyze_contracts()
+
+    @property
+    def detectors(self):
+        return self._detectors
+
+    @property
+    def detectors_high(self):
+        return [d for d in self.detectors if d.IMPACT == DetectorClassification.HIGH]
+
+    @property
+    def detectors_medium(self):
+        return [d for d in self.detectors if d.IMPACT == DetectorClassification.MEDIUM]
+
+    @property
+    def detectors_low(self):
+        return [d for d in self.detectors if d.IMPACT == DetectorClassification.LOW]
+
+    @property
+    def detectors_informational(self):
+        return [d for d in self.detectors if d.IMPACT == DetectorClassification.INFORMATIONAL]
+
+    def register_detector(self, detector_class):
+        """
+        :param detector_class: Class inheriting from `AbstractDetector`.
+        """
+        self._check_common_things('detector', detector_class, AbstractDetector, self._detectors)
+
+        instance = detector_class(self, logger_detector)
+        self._detectors.append(instance)
+
+    def register_printer(self, printer_class):
+        """
+        :param printer_class: Class inheriting from `AbstractPrinter`.
+        """
+        self._check_common_things('printer', printer_class, AbstractPrinter, self._printers)
+
+        instance = printer_class(self, logger_printer)
+        self._printers.append(instance)
+
+    def run_detectors(self):
+        """
+        :return: List of registered detectors results.
+        """
+
+        return [d.detect() for d in self._detectors]
+
+    def run_printers(self):
+        """
+        :return: List of registered printers outputs.
+        """
+
+        return [p.output(self.filename) for p in self._printers]
+
+    def _check_common_things(self, thing_name, cls, base_cls, instances_list):
+
+        if not issubclass(cls, base_cls) or cls is base_cls:
+            raise Exception(
+                "You can't register {!r} as a {}. You need to pass a class that inherits from {}".format(
+                    cls, thing_name, base_cls.__name__
+                )
+            )
+
+        if any(isinstance(obj, cls) for obj in instances_list):
+            raise Exception(
+                "You can't register {!r} twice.".format(cls)
+            )
+
+    def _init_from_vyper(self, contract, **kwargs):
+        vyper = kwargs.get('vyper', 'vyper')
+        contract_json = self._run_vyper(contract, vyper)
+        self._parse_contracts_from_json(contract_json)
+
+        f = open(contract)
+        kode = f.read()
+        code = parse_to_ast(kode)
+        global_ctx = GlobalContext.get_global_context(code)
+        self._global_ctx = global_ctx
+
+    def _run_vyper(self, filename, vyper):
+        if not os.path.isfile(filename):
+            logger.error('{} does not exist (are you in the correct directory?)'.format(filename))
```

Comment:
```suggestion
            logger.error('%s does not exist (are you in the correct directory?)', filename)
```

Reply:
I actually prefer to use the `format` syntax. Any objections for it?